pilot/pkg/model/telemetry.go

```diff
@@ -933,6 +1006,125 @@ var waypointStatsConfig = protoconv.MessageToAny(&udpa.TypedStruct{
 // STOP. DO NOT UPDATE THIS WITHOUT UPDATING buildHTTPTelemetryFilter and buildTCPTelemetryFilter.
 const telemetryFilterHandled = 14
 
+func ECDSResourceName(protocol networking.ListenerProtocol, class networking.ListenerClass, nodeType NodeType) string {
+	p := "HTTP"
+	if protocol == networking.ListenerProtocolTCP {
+		p = "TCP"
+	}
+
+	metricClass := "client"
+	if class == networking.ListenerClassSidecarInbound {
+		metricClass = "server"
+	}
+
+	node := ""
+	if nodeType == Waypoint {
+		node = "waypoint"
+	}
+
+	return fmt.Sprintf("stats|%s|%s|%s", p, metricClass, node)
```

Comment:
I am a bit concerned with this. If we externalize the key from internal (in istiod memory) to external (in envoy), we limit or ability to be flexible here. For example, maybe we want to attach Telemetry at more granular levels

Reply:
I suppose we can extend it in the future but it gets tricky

Reply:
Agree, can we put a URL here? E.g. `istio.io/telemetry/...`

Reply:
I agree this's not good, need a design for ecds resource name.